Adding doc for hub cluster template support (#2753)
open-cluster-management/rhacm-docs: governance/custom_template.adoc

[#support-templates-in-config-policies]
= Support for templates in configuration policies

> Configuration policies support the inclusion of Golang text templates in the object definitions. These templates are resolved at runtime on the target managed cluster using configurations related to that cluster. This gives you the ability to define configuration policies with dynamic content, and inform or enforce Kubernetes resources that are customized to the target cluster. {product-title} supports both hub cluster and managed cluster templates.

Reviewer: ```suggestion
Configuration policies support the inclusion of Golang text templates in the object definitions. These templates are resolved at runtime either on the hub cluster or the target managed cluster using configurations related to that cluster. This gives you the ability to define configuration policies with dynamic content, and inform or enforce Kubernetes resources that are customized to the target cluster.
```

        {{ (fromConfigMap "site-config" "site1" "enabled")  | toBool }}
----

[#hub-templates]
== Support for hub cluster templates in configuration policies

> {product-title-short} also supports hub cluster templates to define configuration policies that are dynamically customized to the target cluster. This prevents the creation of separate policies for each target cluster or hardecodes configuration values in the policy definitions. 

Reviewer: ```suggestion
{product-title-short} also supports hub cluster templates to define configuration policies that are dynamically customized to the target cluster. This reduces the need to create separate policies for each target cluster or hardcode configuration values in the policy definitions. 
```

> Hub cluster templates are based on Golang text template specifications, and uses the `{{hub … hub}}` delimiter for hub cluster templates.

Reviewer: ```suggestion
Hub cluster templates are based on Golang text template specifications, and the `{{hub … hub}}` delimiter indicates a hub cluster template in a configuration policy.
```

[#special-annotation-processing]
=== Special annotation for reprocessing

Policies are processed on the hub cluster only upon creation or after an update. Therefore, hub cluster templates are only resolved to the data in the referenced resources upon policy creation or update. Any changes to the referenced resources are not automatically synced to the policies. 

A special annotation, `policy.open-cluster-management.io/trigger-update` can be used to indicate changes to the data referenced by the templates. Any change to the special annotation value initiates template processing, and the latest contents of the referenced resource are read and updated in the policy definition that is the propagator for processing on managed clusters. A typical way to use this annotation is to increment the value by one each time.

See the following table for a comparison of hub cluster and managed cluster templates:

.Comparison table of hub cluster and managed cluster
|===
| Templates | Hub cluster | Managed cluster 

| Syntax
| Golang text template specification
| Golang text template specification

| Delimiter
| {{hub … hub}}
| {{ … }}

| Functions

> | Set of generic Kubernetes resource such as `lookup` resource, specific `lookup` resources that can be utilized. See <<template-functions,Template functions>> for more information.

Reviewer: ```suggestion
| Set of generic Kubernetes resource such as `lookup` resource, specific `lookup` resources that can be utilized. See <<template-functions,Template functions>> for more information. Resources referenced in the `lookup` function must be in the same namespace as the policy due to a {product-short-title} security restriction. Due to this, the namespace can be set to an empty string when using these functions.
```

| Set of generic Kubernetes resource such as `lookup` resource, specific `lookup` resources that can be utilized. See <<template-functions,Template functions>> for more information.

| Context

> | A `ManagedClusterName` variable is available, which at runtime, resolves to the name of the target cluster where the policy is propagated.

Reviewer: ```suggestion
| A `.ManagedClusterName` variable is available, which at runtime, resolves to the name of the target cluster where the policy is propagated.
```

| No context variables

| Processing
| Processing occurs at runtime on the hub cluster during propagation of replicated policies to clusters. Policies and the hub cluster templates within the policies are processed on the hub cluster only when templates are created or updated.
| Processing occurs in the `ConfigPolicyController` on the managed cluster. Policies are processed periodically, which automatically updates the resolved object definition with data in the referenced resources.

| Access control
| You can only reference Kuberenetes resources that are in the same namespace as the `Policy` resource.
| You can reference any resource on the cluster.

| Processing errors

> | Errors from the hub cluster templates are displayed as violations on the specific target cluster that is propagated.

Reviewer: ```suggestion
| Errors from the hub cluster templates are displayed as violations on the managed clusters the policy applies to.
```

> | Errors from the managed cluster templates are displayed as violations on the specific target cluster where the violationoccurred.

Reviewer: ```suggestion
| Errors from the managed cluster templates are displayed as violations on the specific target cluster where the violation occurred.
```

Author: Great catch, thank you ! Accepting suggestion
